api_spec.yaml

```diff
@@ -708,28 +709,27 @@ paths:
               schema:
                 oneOf:
                   - type: object
+                    required:
+                      - status
+                      - clientInfo
+                      - authorizationCodes
+                      - issuedTokens
                     properties:
                       status:
                         $ref: '#/components/schemas/statusOK'
-                      redirectUris:
-                        type: array
-                        items:
-                          $ref: '#/components/schemas/oauth2RedirectUri'
-                      name:
-                        type: string
-                        example: asdf
-                      scopes:
-                        type: array
-                        items:
-                          $ref: '#/components/schemas/oauth2Scope'
-                      timeCreated:
-                        $ref: '#/components/schemas/timestampMs'
-                      timeUpdated:
-                        $ref: '#/components/schemas/timestampMs'
+                      clientInfo:
+                        $ref: '#/components/schemas/oauth2ClientInfo'
```

Comment:
i don't think that we store the client secret in the database, but instead, we store the client secret hash. So the output of this API cannot have the clientSecret.

You can make the clientSecret field optional in the oauth2ClientInfo type

Reply:
changes made as below:.
  1. made `clientSecret` as optional in `Oauth2ClientSchema`
  2. made `clientSecreat` as required in create oauth2 Client api response